src/analysis/inner_product_space/projection.lean

```diff
@@ -123,7 +123,8 @@ begin
         repeat {exact le_of_lt one_half_pos},
         exact add_halves 1 },
     have eq₁ : 4 * δ * δ ≤ 4 * ∥u - half • (wq + wp)∥ * ∥u - half • (wq + wp)∥,
-    { mono, mono, norm_num, apply mul_nonneg, norm_num, exact norm_nonneg _ },
+    { simp_rw mul_assoc,
+      exact mul_le_mul_of_nonneg_left (mul_self_le_mul_self zero_le_δ eq) zero_le_four },
```

Comment:
As before

Reply:
This use was bad because `apply` and `exact` (which are rigid tactics) were used after `mono` (which is a fancy tactic). All that happened is that the side goals were swapped and the rigid invokations broke.

Reply:
But now `mono` isn't used to close the goal at all, we have to resort to a less automatic tactic - just saying that the old uses were bad isn't enough to justify this